src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -1536,6 +1536,168 @@ int SketchObject::delConstraintOnPoint(int GeoId, PointPos PosId, bool onlyCoinc
+    // to move unpredictably.
+    if (!arePointsCoincident(geoId1, posId1, geoId2, posId2)) {
+        // Delete constraints on the endpoints
+        delConstraintOnPoint(geoId1, posId1, false);
+        delConstraintOnPoint(geoId2, posId2, false);
+
+        // Delete line length and equal constraints
+        const std::vector<Constraint *> &constraints = this->Constraints.getValues();
+        std::vector<int> deleteme;
+        for (int i=0; i < int(constraints.size()); i++) {
+            const Constraint *c = constraints[i];
+            if (c->Type == Sketcher::Distance || c->Type == Sketcher::Equal) {
+                bool line1 = c->First == geoId1 && c->FirstPos == none;
+                bool line2 = c->First == geoId2 && c->FirstPos == none;
+                if (line1 || line2) {
+                  deleteme.push_back(i);
+                }
+            }
+        }
+        delConstraints(deleteme, false);
+        return;
+    }
+
+    // If the lines aren't straight, don't try to manage the constraints.
+    // TODO: Add support for curved lines.
+    const Part::Geometry *geo1 = getGeometry(geoId1);
+    const Part::Geometry *geo2 = getGeometry(geoId2);
+    if (geo1->getTypeId() != Part::GeomLineSegment::getClassTypeId() ||
+        geo2->getTypeId() != Part::GeomLineSegment::getClassTypeId() ) {
+        delConstraintOnPoint(geoId1, posId1, false);
+        delConstraintOnPoint(geoId2, posId2, false);
+        return;
+    }
+
+    // Add a vertex to preserve the original intersection of the filleted lines
+    Part::GeomPoint *originalCorner = new Part::GeomPoint(getPoint(geoId1, posId1));
+    int originalCornerId = addGeometry(originalCorner);
+    delete originalCorner;
+
+    // Constrain the vertex to the two lines
+    Sketcher::Constraint *cornerToLine1 = new Sketcher::Constraint();
+    cornerToLine1->Type = Sketcher::PointOnObject;
+    cornerToLine1->First = originalCornerId;
+    cornerToLine1->FirstPos = start;
+    cornerToLine1->Second = geoId1;
+    cornerToLine1->SecondPos = none;
+    addConstraint(cornerToLine1);
+    delete cornerToLine1;
+    Sketcher::Constraint *cornerToLine2 = new Sketcher::Constraint();
+    cornerToLine2->Type = Sketcher::PointOnObject;
+    cornerToLine2->First = originalCornerId;
+    cornerToLine2->FirstPos = start;
+    cornerToLine2->Second = geoId2;
+    cornerToLine2->SecondPos = none;
+    addConstraint(cornerToLine2);
+    delete cornerToLine2;
+
+    Base::StateLocker lock(managedoperation, true);
+
+    //Base::Console().Log("\nLooking for %d %d and %d %d\n", geoId1, posId1, geoId2, posId2);
+
+    // Loop through all the constraints and try to do reasonable things with the affected ones
+    const std::vector<Constraint *> &constraints = this->Constraints.getValues();
+    std::vector<Constraint *> newConstraints;
+    for (int i=0; i< int(constraints.size()); i++) {
+        Constraint *c = constraints[i];

```

Comment:
I suggest moving to a range-based `for` loop here, it's more compact and less error-prone:
```suggestion
    std::vector<Constraint *> newConstraints;
    for (auto c : this->Constraints.getValues()) {
```

Reply:
C++11 is starting to look downright *modern*.   Done.

Reply:
Have you been following C++20? Concepts, Ranges, _and_ Modules? It's like getting a new language for Christmas.